lib/HLSL/DxilContainerReflection.cpp

```diff
@@ -2324,6 +2326,34 @@ void DxilLibraryReflection::AddResourceDependencies() {
   }
 }
 
+static void CollectCBufUsageForLib(Value *V, std::vector<unsigned> &cbufUsage) {
+  for (auto user : V->users()) {
+    Value *V = user;
+    if (auto *CI = dyn_cast<CallInst>(V)) {
+      if (hlsl::OP::IsDxilOpFuncCallInst(CI, hlsl::OP::OpCode::CreateHandleForLib)) {
+        CollectCBufUsage(CI, cbufUsage);
+      }
+    } else if (isa<GEPOperator>(V) ||
+               isa<LoadInst>(V)) {
+      CollectCBufUsageForLib(user, cbufUsage);
+    }
+  }
+}
+
+void DxilLibraryReflection::SetCBufferUsage() {
+  unsigned cbSize = std::min(m_CBs.size(), m_pDxilModule->GetCBuffers().size());
+  std::vector< std::vector<unsigned> > cbufUsage(cbSize);
```

Comment:
Tiny nit: no need to have more than one vector alive at a time.

Reply:
Your right, that’s what happens when you start wtith another method, then reduce it to this, but forget to review the opportunities afterwards.